src/main/java/org/elasticsearch/transport/TransportInfo.java

```diff
@@ -55,6 +63,16 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.startObject(Fields.TRANSPORT);
         builder.field(Fields.BOUND_ADDRESS, address.boundAddress().toString());
         builder.field(Fields.PUBLISH_ADDRESS, address.publishAddress().toString());
+        if (profileAddresses.size() > 0) {
```

Comment:
How do you feel about always emitting a `profiles` map, that map just be empty? I think it's a cleaner implementation than a key that sometimes can be missing depending on the profile map, it makes client diagnostics easier also (ie, you know where to look, even if there isn't anything there yet)